src/render.rs

```diff
@@ -1,145 +1,343 @@
-//! Render message to a byte buffer
-
 use midi_types::{status::*, MidiMessage};
 
-/// Trait for rendering a MidiMessage into a byte slice.
-pub trait MidiRenderSlice: Sized {
-    /// Render to a slice.
-    ///
-    /// # Panics
+/// This trait abstracts the transport mechanism for the MidiRenderer. An instance of a type that implements this trait can be used by the MidiRenderer to write midi-messages
+pub trait MidiTransport {
+    type Error;
+
+    /// Write a message as series of bytes to the midi transport layer
     ///
-    /// Panics if the slice length is less than 3.
-    fn render_slice(&self, buf: &mut [u8]) -> usize;
+    /// For compatibility this should always be used to write one whole midi-message with a maximum of 4 bytes
+    fn write(&mut self, bytes: &[u8]) -> Result<(), Self::Error>;
```

Comment:
do any of the messages render more than 3 bytes?

Reply:
Not right now, but I think for USB 4 is the maximum? I hope we can implement sysex next and I think that require data-packets of length 4 (1 status + 3 data) for USB.
I see that I did mess this up in the tests on line 328. There is an off-by-one error there. I'll check the USB midi specs again just to be sure and fix that.

Reply:
USB-MIDI adds a header byte but that kind of feels out of scope for the rendering here.. the `usbd-midi` implementation handles that header itself.

Reply:
Ah, yes, I missed that thanks, fixed!